pkg/metrics/server.go

```diff
@@ -0,0 +1,48 @@
+package metrics
+
+import (
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/etc"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+type Server struct {
+	cfg    etc.Metrics
+	server *http.Server
+}
+
+func NewServer(cfg etc.Metrics) *Server {
+	mux := http.NewServeMux()
+	mux.Handle(cfg.Endpoint, promhttp.Handler())
+	return &Server{
+		cfg: cfg,
+		server: &http.Server{
+			Addr:    cfg.Addr,
+			Handler: mux,
+		},
+	}
+}
+
+func (s *Server) ListenAndServe() {
+	go func() {
+		if err := s.listenAndServe(); err != http.ErrServerClosed {
+			log.Fatalf("Error: %v", err)
```

Comment:
If the metrics server shuts down, the adapter process also shuts down. If the metrics server is integrated into API server, we can avoid it.

Reply:
I wanted to keep API & metrics servers separate in order to keep metrics accessible to the infrastructure services, e.g. Prometheus. Since the API server is not secured we can simply and used the same API server.

Reply:
I'm sorry, but I don't understand well yet 😭 
We can add an endpoint "/metrics" so that we can keep metrics accessible from Prometheus.

```
router := mux.NewRouter()
router.Use(handler.logRequest)

apiV1Router := router.PathPrefix("/api/v1").Subrouter()
apiV1Router.Methods(http.MethodPost).Path("/scan").HandlerFunc(handler.AcceptScanRequest)
apiV1Router.Methods(http.MethodGet).Path("/scan/{scan_request_id}/report").HandlerFunc(handler.GetScanReport)
apiV1Router.Methods(http.MethodGet).Path("/metadata").HandlerFunc(handler.GetMetadata)

probeRouter := router.PathPrefix("/probe").Subrouter()
probeRouter.Methods(http.MethodGet).Path("/healthy").HandlerFunc(handler.GetHealthy)
probeRouter.Methods(http.MethodGet).Path("/ready").HandlerFunc(handler.GetReady)

metricsRouter := router.PathPrefix(cfg.Endpoint).Subrouter()
metricsRouter.Methods(http.MethodGet).Path("/").HandleFunc(promhttp.Handler()
```
I'm not sure that this above is correct. This is just an example. Why can't we add the metrics endpoint to API server?

Reply:
We can do that. And I'm going to do that :) In the beginning I was concerned that metrics, which might contain sensitive data. By having a dedicated port / HTTP server we could prevent metrics from being accessed without intercepting HTTP requests.

Reply:
I understand